src/Uno.UI.Composition/Composition/CompositionSpriteShape.skia.cs

```diff
@@ -39,10 +39,11 @@ internal override void Paint(in Visual.PaintingSession session)
 					{
 						// workaround until SkiaSharp adds support for SaveLayerRec
 						fillPaint.FilterQuality = SKFilterQuality.High;
-						session.Canvas.SaveLayer(fillPaint);
-						session.Canvas.Scale(1.0f / session.Canvas.TotalMatrix.ScaleX);
-						session.Canvas.DrawSurface(session.Surface, new(-session.Canvas.TotalMatrix.TransX, -session.Canvas.DeviceClipBounds.Top + session.Canvas.LocalClipBounds.Top));
-						session.Canvas.Restore();
+						var sessionCanvas = session.Surface.Canvas;
```

Comment:
With effect brushes specifically, we have to bypass the `RecorderCanvas` and draw directly on the surface canvas. This is because effect brushes usually after an effect to an existing drawing, and this effect won't appear if you use the recording canvas. I will add a comment.